src/main/java/it/gov/pagopa/debtposition/service/payments/PaymentsService.java

```diff
@@ -412,69 +537,151 @@ private Transfer updateTransferStatus(PaymentPosition pp, String iuv, String tra
   }
   
   /**
-  * Recompute PaymentPosition status (V3 semantics):
-  * - A "plan" is:
-  *   - a group of POs with the same paymentPlanId (isPartialPayment = TRUE), or
-  *   - a single non-installment PO (isPartialPayment = FALSE).
-  * - The PP becomes:
-  *   - REPORTED if at least one plan is fully REPORTED;
-  *   - otherwise PAID if at least one plan is fully {PAID|REPORTED};
-  *   - otherwise PARTIALLY_PAID if a subset of installments is paid/reported;
-  *   - otherwise the status remains unchanged.
-  */
-  private void recomputePaymentPositionStatus(PaymentPosition pp) {
-    Map<String, List<PaymentOption>> groups = pp.getPaymentOption().stream()
-        .collect(Collectors.groupingBy(po -> {
-          if (Boolean.TRUE.equals(po.getIsPartialPayment())) {
-            return "PLAN:" + po.getPaymentPlanId();
-          } else {
-            // each non-installment PO is a single "plan"
-            return "SINGLE:" + po.getIuv();
-          }
-        }));
-
-    boolean anyPlanInProgress = false;    // there is at least one installment paid/reported/partially reported 
-    boolean anyPlanFullyPaid = false;     // there is a fully {PAID|REPORTED} plan
-    boolean anyPlanFullyReported = false; // there is a fully REPORTED plan
-
-    for (List<PaymentOption> group : groups.values()) {
-      boolean allPaidOrReported = true;
-      boolean allReported = true;
-      boolean groupHasAnyProgress = false;
-
-      for (PaymentOption po : group) {
-     
-        if (po.getStatus() == PaymentOptionStatus.PO_PAID
-            || po.getStatus() == PaymentOptionStatus.PO_REPORTED
-            || po.getStatus() == PaymentOptionStatus.PO_PARTIALLY_REPORTED) {
-          groupHasAnyProgress = true;
-        }
+   * Recompute PaymentPosition status (V3 semantics):
+   * - A "plan" is:
+   *   - a group of POs with the same paymentPlanId (isPartialPayment = TRUE), or
+   *   - a single non-installment PO (isPartialPayment = FALSE).
+   * - The PP becomes:
+   *   - REPORTED if at least one plan is fully REPORTED;
+   *   - otherwise PAID if at least one plan is fully {PAID|REPORTED};
+   *   - otherwise PARTIALLY_PAID if a subset of installments is paid/reported;
+   *   - otherwise the status remains unchanged.
+   */
+   private void recomputePaymentPositionStatus(PaymentPosition pp) {
```

Comment:
This code is already on the main in another method, maybe it could be factorized in a single method?

Reply:
ok, so it is view an add for the re-format.